src/common/commonutils/SshUtils.c

```diff
@@ -559,111 +573,214 @@ int CheckSshProtocol(char** reason, void* log)
         {
-            OsConfigLogInfo(log, "SetSshOption: failed setting '%s' to '%s' in '%s' (%d)", option, value, g_sshServerConfiguration, status);
+            status = errno ? errno : ENOENT;
+            OsConfigLogError(log, "SetSshWarningBanner: failed to save banner text '%s' to file '%s' with %d", bannerText, etcAzSec, status);
         }
     }
-    else
-    {
-        OsConfigLogError(log, "SetSshOption: out of memory");
-        status = ENOMEM;
-    }
-
-    FREE_MEMORY(commandResult);
-    FREE_MEMORY(command);
-
-    OsConfigLogInfo(log, "SetSshOption('%s' to '%s'): %s (%d)", option, value, PLAIN_STATUS_FROM_ERRNO(status), status);
 
     return status;
 }
 
-static int SetSshWarningBanner(unsigned int desiredBannerFileAccess, const char* bannerText, void* log)
+static int IncludeSshRemediationConfFile(void* log)
 {
-    const char* etcAzSec = "/etc/azsec/";
+    const char* confFolder = "/etc/ssh/sshd_config.d";
+    const char* configurationTemplate = "%s%s";
+    
+    int desiredAccess = atoi(g_desiredPermissionsOnEtcSshSshdConfig ? g_desiredPermissionsOnEtcSshSshdConfig : g_sshDefaultSshSshdConfigAccess);
+    char* originalConfiguration = NULL;
+    char* newConfiguration = NULL;
+    size_t newConfigurationSize = 0;
     int status = 0;
-       
-    if (NULL == bannerText)
+
+    if (false == FileExists(g_sshServerConfiguration))
     {
-        OsConfigLogError(log, "SetSshWarningBanner: invalid argument");
-        return EINVAL;
+        OsConfigLogInfo(log, "IncludeSshRemediationConfFile: the OpenSSH Server configuration file '%s' is not present on this device", g_sshServerConfiguration);
+        return EEXIST;
     }
-    
-    if (false == DirectoryExists(etcAzSec))
+
+    if (false == DirectoryExists(confFolder))
     {
-        if (0 != mkdir(etcAzSec, desiredBannerFileAccess))
+        if (0 != mkdir(confFolder, desiredAccess))
         {
             status = errno ? errno : ENOENT;
-            OsConfigLogError(log, "SetSshWarningBanner: mkdir(%s, %u) failed with %d", etcAzSec, desiredBannerFileAccess, status);
+            OsConfigLogError(log, "IncludeSshRemediationConfFile: mkdir(%s, %u) failed with %d", confFolder, desiredAccess, status);
         }
     }
 
-    if (true == DirectoryExists(etcAzSec))
+    if (true == DirectoryExists(confFolder))
     {
-        if (SavePayloadToFile(g_sshBannerFile, bannerText, strlen(bannerText), log))
+        if (NULL != (originalConfiguration = LoadStringFromFile(g_sshServerConfiguration, false, log)))
         {
-            if (0 == (status = SetFileAccess(g_sshBannerFile, 0, 0, desiredBannerFileAccess, log)))
+            if (0 != strncmp(originalConfiguration, g_remediationConfHeader, strlen(g_remediationConfHeader)))
             {
-                status = SetSshOption(g_sshBanner, g_sshEscapedBannerFilePath, log);
+                newConfigurationSize = strlen(configurationTemplate) + strlen(g_remediationConfHeader) + strlen(originalConfiguration);
```

Comment:
Is this size guaranteed to account for null terminator?

Reply:
Good catch.